src/utils/claim.ts

```diff
@@ -0,0 +1,116 @@
+import { ClaimDetails, detectSwap } from "boltz-core";
+import { LiquidClaimDetails } from "boltz-core/dist/lib/liquid";
+import log from "loglevel";
+
+import { RBTC } from "../consts";
+import {
+    decodeAddress,
+    getConstructClaimTransaction,
+    getNetwork,
+    getOutputAmount,
+    getTransaction,
+    setup,
+} from "./compat";
+import { ECPair } from "./ecpair";
+import { fetcher, parseBlindingKey } from "./helper";
+
+const createAdjustedClaim = <
+    T extends
+        | (ClaimDetails & { blindingPrivateKey?: Buffer })
+        | LiquidClaimDetails,
+>(
+    swap: any,
+    claimDetails: T[],
+    destination: Buffer,
+    assetHash?: string,
+    blindingKey?: Buffer,
+) => {
+    const inputSum = claimDetails.reduce(
+        (total: number, input: T) => total + getOutputAmount(swap.asset, input),
+        0,
+    );
+    const feeBudget = Math.floor(inputSum - swap.receiveAmount);
+
+    const constructClaimTransaction = getConstructClaimTransaction(swap.asset);
+    return constructClaimTransaction(
+        claimDetails as ClaimDetails[] | LiquidClaimDetails[],
+        destination,
+        feeBudget,
+        true,
+        assetHash,
+        blindingKey,
+    );
+};
+
+export const claim = async (
+    swap: any,
+    swapStatusTransaction: { hex: string },
+    callback: (swap: any) => any,
```

Comment:
We should get rid of the callbacks. ***Especially*** in async functions

Reply:
agreed, next pr